### ISSUE TITLE
Could not insert bucket: ttlSeconds should be >= 0, got -1794967295

### ISSUE THREAD

**Issue by grglzrv** [traefik/traefik#8235]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

I am trying to enable ratelimit but I am getting this error below when I open my website https://pihole.example.com

errror msg:

could not insert bucket

### What did you see instead?

Instead, to see Pihole admin page, I saw this error - could not insert bucket

### What version of Traefik are you using?

```console
$ docker run traefik version
Version:      2.4.9
Codename:     livarot
Go version:   go1.16.5
Built:        2021-06-21T16:17:58Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration?

```yaml
  pi-hole:
    container_name: pi-hole
    image: pihole/pihole:v5.7
    hostname: pihole
    restart: unless-stopped
    # Web panel will be at http://<your pi's ip>:8000 Set your devices or routers DNS to your pi's ip
    ports:
      - "8000:80/tcp"
      - "53:53/tcp"
      - "53:53/udp"
    environment:
      - ServerIP=10.10.50.3
      - DNS1='10.10.50.2#5054'
      - DNS2=''
      - IPv6=false
      - TZ=UTC
      - DNSMASQ_LISTENING=all
    # Replace PASSWORD with a password for the pihole admin panel
      - WEBPASSWORD=PASSWORD
    networks:
      web:
      pihole:
        ipv4_address: 10.10.50.3
    # Config files are bound to /home/pi/pihone on your raspberrypi, if your using a different device other than a raspberry pi change it to somewhere else
    volumes:
      - './etc-pihole/:/etc/pihole/'
      - './etc-dnsmasq.d/:/etc/dnsmasq.d/'
      - './youTube_ads_4_pi-hole/:/root/youTube_ads_4_pi-hole/'
    # This is for the pihole container dns you dont need to change it
    dns:
      - 127.0.0.1
      - 1.1.1.1
    cap_add:
      - NET_ADMIN
    labels:
      - traefik.enable=true
  # Routers
      - traefik.http.routers.pi-hole-https.rule=Host(`pihole.domain.com`)
      - traefik.http.routers.pi-hole-https.middlewares=pihole-compress,pihole-ratelimit
      - traefik.http.routers.pi-hole-https.tls=true
  # Compress Middleware
      - traefik.http.middlewares.pihole-compress.compress=true
  # Rate Limit Middleware
      - traefik.http.middlewares.pihole-ratelimit.ratelimit.average=120
      - traefik.http.middlewares.pihole-ratelimit.ratelimit.period=1m
  # Services
      - traefik.http.services.pi-hole-https.loadbalancer.server.port=80

```

Dynamic config file

```yaml
http:
  routers:
    traefik:
      rule: "Host(`traefik.domain.com`)"
      service: "api@internal"
      middlewares: "auth"
      tls:
        domains:
          - main: "domain.com"
            sans:
              - "*.domain.com"
          - main: "ldomain.com"
            sans:
              - "*.domain.com"
  middlewares:
    auth:
      basicAuth:
        usersFile: "/etc/traefik/usersfile"

tls:
  certificates:
    - certFile: "/etc/certs/fullchain.pem"
      keyFile: "/etc/certs/privkey.pem"
```

```yaml
version: '3'

services:
  traefik:
    image: traefik:latest
    container_name: traefik
    restart: unless-stopped
#    security_opt:
#      - no-new-privileges:true
    ports:
      # Web
      - 80:80
      - 443:443
    networks:
      - web
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      # Map the static conf into the container
      - ./traefik/traefik.yml:/etc/traefik/traefik.yml:ro
      # Map the dynamic conf into the container
      - ./traefik/dynamic_config.yml:/etc/traefik/dynamic_config.yml:ro
      # Map the certificats into the container
      - ./certs:/etc/certs:ro
      # Map the usersfile into the container
      - ./usersfile:/etc/traefik/usersfile
    labels:
      # Enable this container to be mapped by traefik
      # For more information, see: https://docs.traefik.io/providers/docker/#exposedbydefault
      - "traefik.enable=true"
      # Use "traefik" router configuration defined into the dynamic config file: ./traefik/config.yml
      - "traefik.http.routers.traefik=true"
      - "traefik.http.middlewares.to-https.redirectscheme.scheme=https"

networks:
  web:
    external: true
```

traefik static config file

```yaml
global:
  sendAnonymousUsage: false

api:
  dashboard: true
#  insecure: true

providers:
  docker:
    endpoint: "unix:///var/run/docker.sock"
    watch: true
    network: web
    exposedByDefault: false

  file:
    filename: /etc/traefik/dynamic_config.yml
    watch: true

log:
  level: INFO
  format: common

accessLog: {}

entryPoints:
  http:
    address: ":80"
    http:
      redirections:
        entryPoint:
          to: https
          scheme: https

  https:
    address: ":443"
```
Add more configuration information here.

### If applicable, please paste the log output in DEBUG level

```
time="2021-06-25T22:09:33Z" level=debug msg="mime: no media type" middlewareName=pihole-compress@docker middlewareType=Compress
time="2021-06-25T22:09:33Z" level=error msg="could not insert bucket: ttlSeconds should be >= 0, got -1794967295" middlewareType=RateLimiterType middlewareName=pihole-ratelimit@docker
192.168.50.1 - - [25/Jun/2021:22:09:33 +0000] "GET /admin/ HTTP/2.0" 500 24 "-" "-" 108 "pi-hole-https@docker" "-" 0ms
```

**Comment by grglzrv** [comment]
@mpl yes, i am using raspberry pi 4/ 4GB, I provided you the docker run Traefik version from MacBook actually.

here is Traefik version from the Raspberry Pi

`pi@raspberrypi:~ $ docker run traefik version
Version:      2.4.9
Codename:     livarot
Go version:   go1.16.5
Built:        2021-06-21T16:17:58Z
OS/Arch:      linux/arm
pi@raspberrypi:~ $`

**Comment by mpl** [comment]
@grglzrv Even though we haven't been able to reproduce your issue, we have most likely identified the cause of what seems to be an int overflow. However, it's taking a bit more time because we're not sure anymore that the formula used in that part of the code makes total sense, so we're trying to see if we can do better there.

**Comment by grglzrv** [comment]
I have just set up a rate limit to 120 without specifying period time, which means it's 120 per second. I got this ...

![image](https://user-images.githubusercontent.com/19932357/124325312-17236780-db8d-11eb-95db-7af372793e86.png)

When the rate limit is lower value I got the previous error.

**Comment by traefiker** [comment]
Closed by #8254.
